nucypher/policy/conditions/evm.py

```diff
@@ -329,6 +363,78 @@ def __init__(
         self.function_abi = function_abi
         self.contract_function = self._get_unbound_contract_function()
 
+        # call to super must be at the end for proper validation
+        super().__init__(condition_type=condition_type, method=method, *args, **kwargs)
+
+    def _validate_method(self, method):
+        return method
+
+    def _validate_expected_return_type(self):
+        output_abi_types = self._get_abi_types(self.contract_function.contract_abi[0])
+        comparator_value = self.return_value_test.value
+        comparator_index = self.return_value_test.index
+        index_string = (
+            f"@index={comparator_index}" if comparator_index is not None else ""
+        )
+        failure_message = (
+            f"Invalid return value comparison type '{type(comparator_value)}' "
+            f"for '{self.contract_function.fn_name}'{index_string} based on ABI types {output_abi_types}"
+        )
+
+        if len(output_abi_types) == 1:
+            expected_type = output_abi_types[0]
+            if comparator_index is not None and self._is_tuple_type(expected_type):
+                type_entries = self._get_tuple_type_entries(expected_type)
+                expected_type = type_entries[comparator_index]
+            self._validate_value_type(expected_type, comparator_value, failure_message)
+        elif len(output_abi_types) > 1:
+            if comparator_index is not None:
+                # only care about indexed entry
+                expected_type = output_abi_types[comparator_index]
+                self._validate_value_type(
+                    expected_type, comparator_value, failure_message
+                )
+                return
+
+            if not isinstance(comparator_value, Sequence):
+                raise InvalidCondition(failure_message)
+
+            if len(output_abi_types) != len(comparator_value):
+                raise InvalidCondition(failure_message)
+
+            for output_abi_type, component_value in zip(
+                output_abi_types, comparator_value
+            ):
+                self._validate_value_type(
+                    output_abi_type, component_value, failure_message
+                )
+        else:
+            raise InvalidCondition(
+                "No outputs for ABI function."
+            )  # should never happen
```

Comment:
This may possibly happen if an invalid or incomplete ABI is provided to construct a condition?

Reply:
✔️